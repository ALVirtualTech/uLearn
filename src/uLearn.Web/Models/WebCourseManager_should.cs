using System;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;

namespace uLearn.Web.Models
{
	[TestFixture]
	public class WebCourseManager_should
	{
		[Test]
		public void load_all_courses()
		{
			var manager = WebCourseManager.Instance;
			var courses = manager.GetCourses().ToList();
//			foreach (var c in courses)
//				PrintCourse(c);
			Assert.IsTrue(courses.Any(c => c.Id == "ForTests"));
		}

		private static void PrintCourse(Course c)
		{
			Console.WriteLine("Course: " + c.Id + " " + c.Title);
			foreach (var s in c.Slides)
				Console.WriteLine(s.Info.UnitName + " " + s.Title);
			Console.WriteLine();
		}

		[Test]
		public void load_ForTestsCourse_slides()
		{
			var manager = WebCourseManager.Instance;
			Approvals.VerifyAll(manager.GetCourse("ForTests").Slides, "Slides");
		}
	}
}