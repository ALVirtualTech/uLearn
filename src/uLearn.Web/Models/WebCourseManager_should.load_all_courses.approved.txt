Courses[0] = Id: AIML, Title: Искусственный интеллект и машинное обучение
Courses[1] = Id: ForTests, Title: Шаблонный курс
Courses[2] = Id: Help, Title: Справка для авторов курсов (xml)
Courses[3] = Id: Linq, Title: Практикум по LINQ
Courses[4] = Id: OOP, Title: Практикум по объектно-ориентированному программированию
Courses[5] = Id: ParProg, Title: Параллельное программирование
Courses[6] = Id: UXR, Title: Юзабилити
